opensrp-core/src/main/java/org/smartregister/util/OpenSRPImageLoader.java

```diff
@@ -493,118 +271,71 @@ public OpenSRPImageLoader setMaxImageSize(int maxImageSize) {
-            maxWidth, int maxHeight) {
-
-        // Find any old image load request pending on this ImageView (in case this view was
-        // recycled)
-        ImageContainer imageContainer =
-                imageView.getTag() != null && imageView.getTag() instanceof ImageContainer
-                        ? (ImageContainer) imageView.getTag() : null;
-
-        // Find image url from prior request
-        String recycledImageUrl = imageContainer != null ? imageContainer.getRequestUrl() : null;
-
-        // If the new requestUrl is null or the new requestUrl is different to the previous
-        // recycled requestUrl
-        if (requestUrl == null || !requestUrl.equals(recycledImageUrl)) {
-            if (imageContainer != null) {
-                // Cancel previous image request
-                imageContainer.cancelRequest();
-                imageView.setTag(null);
-            }
-            if (requestUrl != null) {
-                // Queue new request to fetch image
-                imageContainer = get(requestUrl,
-                        getImageListener(mResources, imageView, placeHolder, mFadeInImage),
-                        maxWidth, maxHeight);
-                // Store request in ImageView tag
-                imageView.setTag(imageContainer);
-            } else {
-                imageView.setImageDrawable(placeHolder);
-                imageView.setTag(null);
-            }
-        }
+                        AccountAuthenticatorXml authenticatorXml = CoreLibrary.getInstance().getAccountAuthenticatorXml();
+                        String accessToken = AccountHelper.getCachedOAuthToken(CoreLibrary.getInstance().context().allSharedPreferences().fetchRegisteredANM(),
+                                authenticatorXml.getAccountType(),
+                                AccountHelper.TOKEN_TYPE.PROVIDER);
 
-        return imageContainer;
-    }
+                        GlideUrl glideUrl = new GlideUrl(url,
+                                new LazyHeaders.Builder()
+                                        .addHeader(AllConstants.HTTP_REQUEST_HEADERS.AUTHORIZATION, new StringBuilder(AllConstants.HTTP_REQUEST_AUTH_TOKEN_TYPE.BEARER + " ")
+                                                .append(accessToken).toString()).build());
 
-    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
-    protected <T> void startAsyncTask(AsyncTask<T, ?, ?> asyncTask, T[] paramsArg) {
-        T[] params = paramsArg;
+                        Glide.with(context).load(glideUrl)
+                                .transition(DrawableTransitionOptions.withCrossFade(mFadeInImage ? AllConstants.IMAGE_ANIMATION_FADE_IN_TIME : 0))
+                                .placeholder(opensrpImageListener.getDefaultImageResId())
+                                .error(opensrpImageListener.getErrorImageResId())
+                                .into(opensrpImageListener.getImageView());

```

Comment:
It seems that we'll be either losing the functionality or this will render the blocks for checking the image repository useless. Does glide forever keep this disk cache and can we completely switch to using it or can we add instructions to update the ImageRepository that we have the clients image cached by glide or us

Reply:
This is in use in immunization library

Reply:
Glides manages this by creating a 250MB permanent Cache that follows the LRU caching strategy. This can be customised but I believe the current default configuration works well for our use case.

For more see [Glide cache documentation](https://bumptech.github.io/glide/doc/configuration.html#disk-cache)

Reply:
Cool, then we'll need to update the image repository to flag the image as downloaded and optionally change the directory from the default `cache`